src/main/java/net/minestom/server/inventory/click/Click.java

```diff
@@ -0,0 +1,219 @@
+package net.minestom.server.inventory.click;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+
+/**
+ * A tagged union representing possible clicks from the client.
+ */
+public sealed interface Click {
+
+    /**
+     * Gets the slot of this click. -999 indicates the click either drops the cursor item in some way (implements
```

Comment:
Would be nice to note that the slot is the slot "relative" (maybe a better word) to the inventory it was clicked in irrespective of what gui size is currently open.